### PR TITLE
Pull out calculation functions from disease observer

### PR DESCRIPTION
**Pull request by collijk**
Tests on the way

### CODE REVIEW

File: tests/metrics/test_utilities.py [hunk 1]
```diff
@@ -46,3 +78,66 @@ def test_query_string(a, b):
...
+    expected_value = len(pop) / 2
+    if observer_config['by_sex']:
+        expected_value /= 2
+    if observer_config['by_age']:
+        expected_value /= len(age_bins)
+
+    # Doubling pop should double person time
+    pop = pd.concat([pop, pop], axis=0, ignore_index=True)
+
+    person_time = get_disease_event_counts(pop, observer_config, disease, event_time, age_bins)
```
Comment (kfwilson) [inline tests/metrics/test_utilities.py]: this should probably be counts instead of person_time?

File: tests/metrics/test_utilities.py [hunk 2]
```diff
@@ -1,8 +1,40 @@
...
+
+
+@pytest.fixture(params=((0, 100, 5, 1000), (20, 100, 5, 1000)))
+def ages_and_bins(request):
+    age_min = request.param[0]
+    age_max = request.param[1]
+    age_groups = request.param[2]
+    num_ages = request.param[3]
+
+    ages = np.linspace(age_min, age_max - age_groups/num_ages, num_ages)
```
Comment (kfwilson) [inline tests/metrics/test_utilities.py]: ooh I like this. So much better than manually making up data (my bad goto!)

File: src/vivarium_public_health/metrics/utilities.py [hunk 3]
```diff
@@ -80,18 +80,18 @@ def get_output_template(by_age: bool, by_sex: bool, by_year: bool) -> Template:
         A template string with measure and possibly additional criteria.
 
     """
-    template = '{measure}'
+    template = '${measure}'
```
Comment (kfwilson) [inline src/vivarium_public_health/metrics/utilities.py]: I haven't really used template strings before so why do you need the $?

Comment (collijk) [reply src/vivarium_public_health/metrics/utilities.py]: Holdover from old string formatting, I think.  I was surprised too.  It's just the escape character used here.  I think we could customize that, but not worth it.

File: tests/metrics/test_utilities.py (hunk 2 above)
Comment (collijk) [reply tests/metrics/test_utilities.py]: Building data is always hard.

File: tests/metrics/test_utilities.py (hunk 1 above)
Comment (collijk) [reply tests/metrics/test_utilities.py]: Will fix in the next PR

**Pull request merged**